test/integration/extensions/controller/backupbucket_test.go

```diff
@@ -0,0 +1,298 @@
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/operation/common"
+	"github.com/gardener/gardener/pkg/utils"
+	"github.com/gardener/gardener/test/framework"
+	"github.com/gardener/gardener/test/integration/extensions/controller/fake"
+	"github.com/gardener/gardener/test/integration/extensions/controller/fake/backupbucket"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	gomegatypes "github.com/onsi/gomega/types"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+var _ = Describe("BackupBucket", func() {
+	AfterEach(func() {
+		framework.RunCleanupActions()
+	})
+
+	It("should successfully create and delete a BackupBucket (ignoring operation annotation)", func() {
+		prepareAndRunTest(true)
+	})
+
+	It("should successfully create and delete a BackupBucket (respecting operation annotation)", func() {
+		prepareAndRunTest(false)
+	})
+})
+
+func prepareAndRunTest(ignoreOperationAnnotation bool) {
+	By("setup and start manager")
+	mgr, err := createAndStartManager(ignoreOperationAnnotation)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("setup namespace for test")
+	namespace, err := generateNamespaceName()
+	Expect(err).NotTo(HaveOccurred())
+
+	runTest(mgr.GetClient(), namespace, ignoreOperationAnnotation)
+}
+
+func createAndStartManager(ignoreOperationAnnotation bool) (manager.Manager, error) {
+	mgrContext, mgrCancel := context.WithCancel(ctx)
+
+	framework.AddCleanupAction(func() {
+		defer func() {
+			By("stopping manager")
+			mgrCancel()
+		}()
+	})
+
+	mgr, err := manager.New(restConfig, manager.Options{})
+	if err != nil {
+		return nil, err
+	}
+
+	if err := extensionsv1alpha1.AddToScheme(mgr.GetScheme()); err != nil {
+		return nil, err
+	}
+
+	if err := backupbucket.AddToManagerWithOptions(mgr, ignoreOperationAnnotation); err != nil {
+		return nil, err
+	}
+
+	go func() {
+		if err := mgr.Start(mgrContext.Done()); err != nil {
+			panic(err)
+		}
+	}()
+
+	return mgr, nil
+}
+
+func generateNamespaceName() (string, error) {
+	suffix, err := utils.GenerateRandomStringFromCharset(5, "0123456789abcdefghijklmnopqrstuvwxyz")
+	if err != nil {
+		return "", err
+	}
+	return "gextlib-backupbucket-it--" + suffix, nil
+}
+
+func runTest(c client.Client, namespaceName string, ignoreOperationAnnotation bool) {
+	var (
+		namespace    *corev1.Namespace
+		cluster      *extensionsv1alpha1.Cluster
+		secret       *corev1.Secret
+		backupBucket *extensionsv1alpha1.BackupBucket
+	)
+
+	framework.AddCleanupAction(func() {
+		By("delete backupbucket")
+		Expect(client.IgnoreNotFound(c.Delete(ctx, backupBucket))).To(Succeed())
```

Comment:
If the test fails before the BB has been created, this step fails and the other cleanup activities are not executed.
Maybe only execute if BB != nil

Reply:
Hm, I don't understand, why does this matter? The test is failed anyways already, so why do we need special care here in the cleanup action? cc @ialidzhikov

Reply:
Because otherwise the namespace is for instance not cleaned up.